VAS.Core/ViewModel/ProjectVM.cs

```
@@ -17,13 +17,87 @@
 //
 using System.Threading.Tasks;
 using VAS.Core.Events;
+using VAS.Core.Interfaces.MVVMC;
 using VAS.Core.MVVMC;
 using VAS.Core.Store;
 
 namespace VAS.Core.ViewModel
 {
-	public class ProjectVM<T> : ViewModelBase<T> where T : Project
+
+	/// <summary>
+	/// A ViewModel for <see cref="Project"/> objects.
+	/// </summary>
+	public class ProjectVM : ViewModelBase<Project>
 	{
+		Project model;
+
+		public ProjectVM ()
+		{
+			Playlists = new PlaylistCollectionVM ();
+			EventTypes = new CollectionViewModel<EventType, EventTypeVM> ();
+			EventTypesTimeline = new EventTypesTimelineVM ();
+		}
+
+		public override Project Model {
+			get {
+				return model;
+			}
+
+			set {
+				model = value;
+				UpdateModels ();
```

if a null Value is Passed to Model it will crash, I don't know if it is possible to pass a null value to Model. But if it isn't possible Let it crash is OK.